provider/source/src/time_zones/convert.rs

```diff
@@ -190,54 +199,174 @@ impl From<CldrTimeZonesData<'_>> for ExemplarCitiesV1<'static> {
+) -> impl Iterator<Item = (String, &Vec<MetazoneForPeriod>)> {
+    data.iter().flat_map(|(key, zone)| match zone {
+        ZonePeriod::Region(periods) => vec![(key.to_string(), periods)],
+        ZonePeriod::LocationOrSubRegion(place) => place
+            .iter()
+            .flat_map(
+                move |(key2, location_or_subregion)| match location_or_subregion {
+                    MetaLocationOrSubRegion::Location(periods) => {
+                        vec![(format!("{key}/{key2}"), periods)]
+                    }
+                    MetaLocationOrSubRegion::SubRegion(subregion) => subregion
+                        .iter()
+                        .flat_map(move |(key3, periods)| {
+                            vec![(format!("{key}/{key2}/{key3}"), periods)]
+                        })
+                        .collect::<Vec<_>>(),
+                },
+            )
+            .collect::<Vec<_>>(),
+    })
+}
+
 impl From<CldrTimeZonesData<'_>> for MetazonePeriodV1<'static> {
     fn from(other: CldrTimeZonesData<'_>) -> Self {
-        let data = other.meta_zone_periods_resource;
         let bcp47_tzid_data = &compute_bcp47_tzids_btreemap(other.bcp47_tzids_resource);
         let meta_zone_id_data = &compute_meta_zone_ids_btreemap(other.meta_zone_ids_resource);
         Self(
-            data.iter()
-                .flat_map(|(key, zone)| match zone {
-                    ZonePeriod::Region(periods) => match bcp47_tzid_data.get(key) {
-                        Some(bcp47) => {
-                            vec![(*bcp47, periods, meta_zone_id_data)]
+            iter_meta_zone_periods(other.meta_zone_periods_resource)
+                .flat_map(|(iana, periods)| match bcp47_tzid_data.get(&iana) {
+                    Some(&bcp47) => metazone_periods_iter(bcp47, periods, meta_zone_id_data),
+                    None => panic!("Cannot find bcp47 for {iana:?}."),
+                })
+                .collect(),
+        )
+    }
+}
+
+impl From<CldrTimeZonesData<'_>> for ZoneOffsetPeriodV1<'static> {
+    fn from(other: CldrTimeZonesData<'_>) -> Self {
+        Self(
+            compute_canonical_tzids_btreemap(other.bcp47_tzids_resource)
+                .into_iter()
+                .filter_map(|(bcp47, iana)| Some((bcp47, other.tzdb.get_zoneset(&iana)?)))
+                .flat_map(|(bcp47, zoneset)| {
+                    let mut data = Vec::<(IsoMinutesSinceEpoch, (UtcOffset, UtcOffset))>::new();
+
+                    let store_offsets =
+                        |data: &mut Vec<(IsoMinutesSinceEpoch, (UtcOffset, UtcOffset))>,
+                         end_time: IsoMinutesSinceEpoch,
+                         mut utc_offset: i64,
+                         mut dst_offset_relative: i64| {
+                            if dst_offset_relative < 0 {
+                                utc_offset += dst_offset_relative;
+                                dst_offset_relative = -dst_offset_relative;
+                            }
```

Comment:
Suggestion: add a comment saying that we do this +/- transformation because CLDR expects it in cases such as Ireland Standard Time `iedub`.

Also maybe add this to the public docs of `ZoneVariant`.